client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteCache.java

```diff
@@ -155,6 +156,15 @@
     */
    NotifyingFuture<Boolean> replaceWithVersionAsync(K key, V newValue, long version, int lifespanSeconds, int maxIdleSeconds);
 
+   /**
+    * Retrieve entries from the server
+    *
+    * @param filterConverterFactory Factory name for the KeyValueFilterConverter or null
+    * @param segments               The segments to iterate on or null if all segments should be iterated
+    * @param batchSize              The number of entries transferred from the server at a time
+    * @return Iterator for the entries
+    */
+   CloseableIterator<Entry<Object, Object>> retrieveEntries(String filterConverterFactory, Set<Integer> segments, int batchSize);
```

Comment:
>  The client however may want to do some sort of partitioning, in which case doing this by segment is a very logical thing.

yes, that was the motivation of the segment based filter.

Wrt to the 'extra' methods, I think regardless of what it may look we agree that  
```CloseableIterator retrieveEntries(String filterConverterFactory, Set<Integer> segments, int batchSize)```  is fine?

I was planning to work in the ```Partitioner``` in a follow up jira, that will carry the extra methods to calculate those segments.

Btw, the ```segments``` param is optional, probably it'd make it more user friendly by exposing 
```CloseableIterator retrieveEntries(String filterConverterFactory, int batchSize)```  as well?


> Do we want another one like getAll but instead returns an iterator?

That might come handy!